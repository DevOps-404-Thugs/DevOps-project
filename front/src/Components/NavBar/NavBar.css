.menu{
    padding: 0 20px;
    border-bottom: solid 1px #e8e8e8;
    overflow: auto;
    box-shadow: 0 0 30px #f3f1f1;
    /* background-color: white; */
}

.menu_container{
    padding: 0px 5px;
}

.menu_container .menu_right{
    float: right;
}

.menu_container .menu_left{
    float: left;
}

