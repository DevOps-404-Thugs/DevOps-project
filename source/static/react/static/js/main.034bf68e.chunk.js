(this.webpackJsonpfront=this.webpackJsonpfront||[]).push([[0],{108:function(e,t,a){},127:function(e,t,a){"use strict";a.r(t);var n=a(0),r=a.n(n),l=a(27),o=a.n(l),s=(a(90),a(17)),c=a(9),i=a(11),u=a(14),m=a.n(u),d=(a(108),a(46)),p=a(42),h=a(81),f=a(30);var g=function(){var e=Object(n.useState)(!1),t=Object(i.a)(e,2),a=t[0],l=t[1];return Object(n.useEffect)((function(){m.a.get("/login",{withCredentials:!0}).then((function(e){console.log("loginGET:"+e.status),console.log(e),200===e.status?l(!0):205===e.status&&l(!1)})).catch((function(e){console.log("loginGET:"+e.response.status),console.log(e.response)}))}),[]),console.log(a),r.a.createElement(d.a,{bg:"light",expand:"lg"},r.a.createElement(d.a.Brand,{href:"/ihomie"},"iHomie"),r.a.createElement(d.a.Toggle,{"aria-controls":"basic-navbar-nav"}),r.a.createElement(d.a.Collapse,{id:"basic-navbar-nav"},r.a.createElement(p.a,{className:"mr-auto"},r.a.createElement(p.a.Link,{href:"/ihomie"},"Home"),r.a.createElement(p.a.Link,{href:"/ihomie#/upload"},"Upload"),r.a.createElement(p.a.Link,{href:"/ihomie#/register"},"Register"),r.a.createElement(p.a.Link,{href:"/ihomie#/account"},"Account")),r.a.createElement(h.a,{inline:!0},!1===a?r.a.createElement(f.a,{variant:"outline-primary",onClick:function(e){window.location="/ihomie#/login"}},"Login"):r.a.createElement(f.a,{variant:"primary",onClick:function(e){m.a.get("/logout",{withCredentials:!0}).then((function(e){console.log("logoutGET:"+e.status),console.log(e),200===e.status&&l(!1)})).catch((function(e){console.log("logoutGET:"+e.response.status),console.log(e.response)}))}},"Logout"))))},E=a(132),b=a(131);var v=function(){var e=Object(n.useState)([]),t=Object(i.a)(e,2),a=t[0],l=t[1];Object(n.useEffect)((function(){m.a.get("/housings").then((function(e){console.log(e.data),console.log("housingsGET:"+e.status),l(e.data)}))}),[]);var o=a.map((function(e,t){var a=Math.abs(e._id.$oid.split("").reduce((function(e,t){return(e=(e<<5)-e+t.charCodeAt(0))&e}),0))%23;return r.a.createElement(E.a,{lg:6,md:8,xs:24,style:{border:"#CCC solid 1px","border-radius":"10px",padding:"30px",margin:"30px auto 30px"}},r.a.createElement("img",{style:{width:"100%"},src:"/static/react/pics/"+a+".jpg",alt:"houseImage"}),r.a.createElement("p",{style:{"padding-top":"10px",margin:"0","font-size":"24px","font-weight":"700"}},e.name),r.a.createElement("p",{style:{"padding-top":"10px","padding-bottom":"5px",margin:"0"}},e.address),r.a.createElement(s.b,{style:{"padding-top":"10px",margin:"0"},to:{pathname:"/detail",state:{objectId:e._id.$oid}}},"Go to see detail"))}));return r.a.createElement("div",{style:{width:"65%",margin:"3rem auto"}},r.a.createElement("div",{style:{textAlign:"center"}},r.a.createElement("h2",null,"  Find Your Future Home in iHomie ")),r.a.createElement("div",null,r.a.createElement(b.a,{gutter:[16,16]},o)),r.a.createElement("br",null),r.a.createElement("br",null),r.a.createElement("div",{style:{display:"flex",justifyContent:"center"}},r.a.createElement(s.b,{className:"link",to:"/upload"},"Upload New House")))},y=a(76),w=a(32),C=a.n(w),j=a(54),S=a(77),x=a(78),O=a(83),k=a(82),T=a(79),I=a(80),B=a(39),A=a(20),N=a(41),P=a(33),H=a(45),F=a(59),L=function(e){Object(O.a)(a,e);var t=Object(k.a)(a);function a(e){var n;return Object(S.a)(this,a),(n=t.call(this,e)).state={name:"",address:"",prevState:null,needLogin:!1,show:!1,showSuccessBar:!1,showFailureBar:!1,errorCode:0,goHomePage:!1},n}return Object(x.a)(a,[{key:"componentDidMount",value:function(){var e=Object(j.a)(C.a.mark((function e(){var t;return C.a.wrap((function(e){for(;;)switch(e.prev=e.next){case 0:return e.next=2,fetch("/housings/".concat(this.props.objectId)).then((function(e){return e.json()}));case 2:t=e.sent,this.setState(t);case 4:case"end":return e.stop()}}),e,this)})));return function(){return e.apply(this,arguments)}}()},{key:"handleInputChange",value:function(e){this.setState(Object(y.a)({},e.target.name,e.target.value))}},{key:"handleModify",value:function(e){this.setState({prevState:this.state,show:!0})}},{key:"handleSave",value:function(){var e=Object(j.a)(C.a.mark((function e(t){var a=this;return C.a.wrap((function(e){for(;;)switch(e.prev=e.next){case 0:return e.next=2,fetch("/housings/".concat(this.props.objectId),{method:"PUT",headers:{"Content-Type":"application/json"},credentials:"include",body:JSON.stringify(this.state)}).then((function(e){405==e.status?a.setState({needLogin:!0}):200==e.status?a.setState({show:!1,showSuccessBar:!0}):(a.state.prevState.showFailureBar=!0,a.state.prevState.errorCode=e.status,a.setState(a.state.prevState))}));case 2:e.sent;case 3:case"end":return e.stop()}}),e,this)})));return function(t){return e.apply(this,arguments)}}()},{key:"handleDelete",value:function(){var e=Object(j.a)(C.a.mark((function e(t){var a=this;return C.a.wrap((function(e){for(;;)switch(e.prev=e.next){case 0:return e.next=2,fetch("/housings/".concat(this.props.objectId),{method:"DELETE",headers:{"Content-Type":"application/json"},credentials:"include"}).then((function(e){405==e.status?a.setState({needLogin:!0}):200==e.status?a.setState({goHomePage:!0}):a.setState({show:!1,showFailureBar:!0,errorCode:e.status})}));case 2:e.sent;case 3:case"end":return e.stop()}}),e,this)})));return function(t){return e.apply(this,arguments)}}()},{key:"handleCancel",value:function(e){this.state.prevState.show=!1,this.setState(this.state.prevState)}},{key:"handleClose",value:function(e){this.setState({show:!1})}},{key:"handleCloseSuccessBar",value:function(e){this.setState({showSuccessBar:!1})}},{key:"handleCloseFailureBar",value:function(e){this.setState({showFailureBar:!1})}},{key:"render",value:function(){var e=this.state,t=e.name,a=e.address,n=e.needLogin,l=e.show,o=e.showSuccessBar,s=e.showFailureBar,i=e.errorCode,u=e.goHomePage;if(n)return r.a.createElement(c.a,{to:"/login"});if(u)return r.a.createElement(c.a,{to:"/"});var m=Math.abs(this.props.objectId.split("").reduce((function(e,t){return(e=(e<<5)-e+t.charCodeAt(0))&e}),0))%23;return r.a.createElement(r.a.Fragment,null,r.a.createElement(F.a,{show:o,variant:"success",onClose:this.handleCloseSuccessBar.bind(this),dismissible:!0},"Modified information successfully!"),r.a.createElement(F.a,{show:s,variant:"danger",onClose:this.handleCloseFailureBar.bind(this),dismissible:!0},"Failed to modify information, error response code: ",i),r.a.createElement(N.a,{show:l,onHide:this.handleClose.bind(this)},r.a.createElement(N.a.Header,{closeButton:!0},r.a.createElement(N.a.Title,null,"Modify house information")),r.a.createElement(N.a.Body,null,r.a.createElement(P.a,{className:"mb-3"},r.a.createElement(P.a.Prepend,null,r.a.createElement(P.a.Text,{id:"basic-addon3"},"Name")),r.a.createElement(H.a,{name:"name","aria-describedby":"basic-addon3",value:t,onChange:this.handleInputChange.bind(this)})),r.a.createElement(P.a,{className:"mb-3"},r.a.createElement(P.a.Prepend,null,r.a.createElement(P.a.Text,{id:"basic-addon3"},"Address")),r.a.createElement(H.a,{name:"address","aria-describedby":"basic-addon3",value:a,onChange:this.handleInputChange.bind(this)}))),r.a.createElement(N.a.Footer,null,r.a.createElement(f.a,{variant:"primary",onClick:this.handleSave.bind(this)},"Save Changes"),r.a.createElement(f.a,{variant:"secondary",onClick:this.handleCancel.bind(this)},"Cancel"))),r.a.createElement(T.a,null,r.a.createElement(I.a,null,r.a.createElement(B.a,null),r.a.createElement(B.a,null,r.a.createElement(A.a,{style:{width:"50em",margin:"1em"}},r.a.createElement(A.a.Img,{variant:"top",src:"/static/react/pics/"+m+".jpg"}),r.a.createElement(A.a.Body,null,r.a.createElement(A.a.Title,null,t),r.a.createElement(A.a.Subtitle,{className:"mb-2 text-muted"},a),r.a.createElement(A.a.Text,null,"A property description is the written portion of a real estate listing that describes the real estate for sale or lease. Nowadays, most buyers begin their property search online. Therefore, real estate descriptions are your best chance to sway buyers and sellers."),r.a.createElement(f.a,{variant:"primary",onClick:this.handleModify.bind(this)},"Modify"),r.a.createElement(f.a,{variant:"primary",onClick:this.handleDelete.bind(this),style:{margin:"1px"}},"Delete")))),r.a.createElement(B.a,null))))}}]),a}(r.a.Component);var D=Object(c.g)((function(e){return r.a.createElement(L,{objectId:e.location.state.objectId})}));var M=function(){var e=Object(n.useState)(""),t=Object(i.a)(e,2),a=t[0],l=t[1],o=Object(n.useState)(""),s=Object(i.a)(o,2),c=s[0],u=s[1];return r.a.createElement("div",null,r.a.createElement("div",{style:{width:"75%",margin:"3rem auto"}},r.a.createElement("form",null,r.a.createElement("div",{class:"form-group"},r.a.createElement("label",null,"House Name"),r.a.createElement("input",{type:"text",onChange:function(e){l(e.currentTarget.value)},class:"form-control",id:"email-input",placeholder:"Input your email address here"})),r.a.createElement("div",{class:"form-group"},r.a.createElement("label",null,"Address"),r.a.createElement("input",{type:"text",onChange:function(e){u(e.currentTarget.value)},class:"form-control",id:"password-input",placeholder:"Input your password here"})),r.a.createElement("button",{type:"submit",class:"btn btn-primary",onClick:function(e){if(e.preventDefault(),!a||!c)return alert("Please fill in all fields!");console.log(a),console.log(c);var t={name:a,address:c};m.a.post("/housings",t,{withCredentials:!0}).then((function(e){console.log(e.status),console.log(e),200===e.status?alert("Upload successfully!"):alert("You meet with an error!")}))}},"Submit"))))};var U=function(){var e=Object(n.useState)(""),t=Object(i.a)(e,2),a=t[0],l=t[1],o=Object(n.useState)(""),c=Object(i.a)(o,2),u=c[0],d=c[1];return r.a.createElement("div",{style:{width:"75%",margin:"3rem auto"}},r.a.createElement("form",null,r.a.createElement("div",{class:"form-group"},r.a.createElement("label",null,"Email address"),r.a.createElement("input",{type:"email",onChange:function(e){l(e.currentTarget.value)},class:"form-control",id:"email-input",placeholder:"Input your email address here"})),r.a.createElement("div",{class:"form-group"},r.a.createElement("label",null,"Password"),r.a.createElement("input",{type:"password",onChange:function(e){d(e.currentTarget.value)},class:"form-control",id:"password-input",placeholder:"Input your password here"})),r.a.createElement("button",{type:"submit",class:"btn btn-primary",onClick:function(e){if(e.preventDefault(),!a||!u)return alert("Please fill in all fields!");var t={email:a,password:u};m.a.post("/login",t,{withCredentials:!0}).then((function(e){200===e.status?(alert("Login successfully!"),window.location.replace("/ihomie")):alert("You meet with an error!")})).catch((function(e){401===e.response.status?alert("You need to register for the account!"):400===e.response.status?alert("Password Error!"):402===e.response.status&&alert("Parameters Error!")}))}},"Submit")),r.a.createElement("small",{class:"text-muted"},"Need An Account? ",r.a.createElement(s.b,{className:"link",to:"/register"},"Sign Up Now")))};var Y=function(){var e=Object(n.useState)(""),t=Object(i.a)(e,2),a=t[0],l=t[1],o=Object(n.useState)(""),c=Object(i.a)(o,2),u=c[0],d=c[1],p=Object(n.useState)(""),h=Object(i.a)(p,2),f=h[0],g=h[1];return r.a.createElement("div",{style:{width:"75%",margin:"3rem auto"}},r.a.createElement("form",null,r.a.createElement("div",{class:"form-group"},r.a.createElement("label",null,"Username"),r.a.createElement("input",{type:"text",onChange:function(e){l(e.currentTarget.value)},class:"form-control",id:"email-input",placeholder:"Input username here"})),r.a.createElement("div",{class:"form-group"},r.a.createElement("label",null,"Email address"),r.a.createElement("input",{type:"email",onChange:function(e){d(e.currentTarget.value)},class:"form-control",id:"email-input",placeholder:"Input your email address here"})),r.a.createElement("div",{class:"form-group"},r.a.createElement("label",null,"Password"),r.a.createElement("input",{type:"password",onChange:function(e){g(e.currentTarget.value)},class:"form-control",id:"password-input",placeholder:"Input your password here"})),r.a.createElement("button",{type:"submit",class:"btn btn-primary",onClick:function(e){if(e.preventDefault(),!a||!u||!f)return alert("Please fill in all fields!");console.log(a),console.log(u),console.log(f);var t={username:a,email:u,password:f};console.log(t),m.a.post("/register",t).then((function(e){console.log(e.status),console.log(e),200===e.status?alert("Register successfully!"):alert("You meet with an error!")})).catch((function(e){console.log(e.response),401===e.response.status?alert("The email has been registered!"):402===e.response.status?alert("This username is unavailable!"):400===e.response.status?alert("You have loged in already!"):403===e.response.status&&alert("Parameter wrong!")}))}},"Submit")),r.a.createElement("small",{class:"text-muted"},"Already Have An Account? ",r.a.createElement(s.b,{className:"link",to:"/login"},"Sign In")))};var G=function(){var e=Object(n.useState)({username:null,password:null}),t=Object(i.a)(e,2),a=t[0],l=t[1];return Object(n.useEffect)((function(){m.a.get("/account").then((function(e){console.log(e.data),l(e.data)})).catch((function(e){console.log(e.response)})),console.log(a)})),r.a.createElement("div",{style:{width:"50%",margin:"3rem auto"}},null==a.username?r.a.createElement("h2",{style:{display:"flex",justifyContent:"center"}},"You need to login first!"):r.a.createElement("div",null,r.a.createElement(A.a,{style:{"margin-bottom":"30px"}},r.a.createElement(A.a.Header,{style:{textAlign:"center"}},"Account Info"),r.a.createElement(A.a.Title,{style:{"padding-top":"0.75rem","padding-left":"1rem"}},"username: ",a.username),r.a.createElement(A.a.Text,{style:{"padding-bottom":"0.75rem","padding-left":"1rem","font-weight":"bold"}},"email: ",a.email)),r.a.createElement(s.b,{className:"link",to:"/account_update",style:{color:"blue",display:"flex",justifyContent:"center"}},"Update Account Information")),r.a.createElement("br",null),r.a.createElement("br",null),r.a.createElement("div",{style:{display:"flex",justifyContent:"center"}}))};var _=function(){var e=Object(n.useState)(""),t=Object(i.a)(e,2),a=t[0],l=t[1],o=Object(n.useState)(""),s=Object(i.a)(o,2),c=s[0],u=s[1];return r.a.createElement("div",null,r.a.createElement("div",{style:{width:"75%",margin:"3rem auto"}},r.a.createElement("form",null,r.a.createElement("div",{class:"form-group"},r.a.createElement("label",null,"User Name"),r.a.createElement("input",{type:"text",onChange:function(e){l(e.currentTarget.value)},class:"form-control",id:"username-input",placeholder:"Input new username here"})),r.a.createElement("div",{class:"form-group"},r.a.createElement("label",null,"Email"),r.a.createElement("input",{type:"text",onChange:function(e){u(e.currentTarget.value)},class:"form-control",id:"email-input",placeholder:"Input new email address here"})),r.a.createElement("button",{type:"submit",class:"btn btn-primary",onClick:function(e){if(e.preventDefault(),!a||!c)return alert("Please fill in all fields!");console.log(a),console.log(c);var t={username:a,email:c};m.a.put("/account",t,{withCredentials:!0}).then((function(e){console.log(e.status),console.log(e),200===e.status?(alert("Account updated successfully!"),window.location.replace("/ihomie")):alert("You meet with an error!")})).catch((function(e){console.log(e.response),400===e.response.status&&alert("This account has existed!")}))}},"Submit"))))};var J=function(){return r.a.createElement("div",{className:"App"},r.a.createElement(s.a,null,r.a.createElement(g,null),r.a.createElement(c.d,null,r.a.createElement(c.b,{exact:!0,path:"/",component:v}),r.a.createElement(c.b,{exact:!0,path:"/detail",component:D}),r.a.createElement(c.b,{exact:!0,path:"/upload",component:M}),r.a.createElement(c.b,{exact:!0,path:"/login",component:U}),r.a.createElement(c.b,{exact:!0,path:"/register",component:Y}),r.a.createElement(c.b,{exact:!0,path:"/account",component:G}),r.a.createElement(c.b,{exact:!0,path:"/account_update",component:_}))))};Boolean("localhost"===window.location.hostname||"[::1]"===window.location.hostname||window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/));a(126);o.a.render(r.a.createElement(r.a.StrictMode,null,r.a.createElement(J,null)),document.getElementById("root")),"serviceWorker"in navigator&&navigator.serviceWorker.ready.then((function(e){e.unregister()})).catch((function(e){console.error(e.message)}))},85:function(e,t,a){e.exports=a(127)},90:function(e,t,a){}},[[85,1,2]]]);
//# sourceMappingURL=main.034bf68e.chunk.js.map