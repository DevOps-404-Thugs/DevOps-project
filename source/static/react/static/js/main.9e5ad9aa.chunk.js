(this.webpackJsonpweb=this.webpackJsonpweb||[]).push([[0],[,,,function(e,a,t){e.exports=t.p+"static/media/logo.5d5d9eef.svg"},function(e,a,t){e.exports=t(11)},,,,,function(e,a,t){},function(e,a,t){},function(e,a,t){"use strict";t.r(a);var s=t(0),n=t.n(s),r=t(1),c=t.n(r),l=(t(9),t(3)),m=t.n(l);t(10);function o(e){return n.a.createElement("div",{className:"preview col-md-6"},n.a.createElement("div",{className:"preview-pic tab-content"},n.a.createElement("div",{className:"tab-pane active",id:"pic-1"},n.a.createElement("img",{src:"https://bootstrapmade.com/demo/themes/EstateAgency/assets/img/slide-2.jpg"})),n.a.createElement("div",{className:"tab-pane",id:"pic-2"},n.a.createElement("img",{src:"https://bootstrapmade.com/demo/themes/EstateAgency/assets/img/slide-2.jpg"})),n.a.createElement("div",{className:"tab-pane",id:"pic-3"},n.a.createElement("img",{src:"https://bootstrapmade.com/demo/themes/EstateAgency/assets/img/slide-2.jpg"})),n.a.createElement("div",{className:"tab-pane",id:"pic-4"},n.a.createElement("img",{src:"https://bootstrapmade.com/demo/themes/EstateAgency/assets/img/slide-2.jpg"})),n.a.createElement("div",{className:"tab-pane",id:"pic-5"},n.a.createElement("img",{src:"https://bootstrapmade.com/demo/themes/EstateAgency/assets/img/slide-2.jpg"}))),n.a.createElement("ul",{className:"preview-thumbnail nav nav-tabs"},n.a.createElement("li",{className:"active"},n.a.createElement("a",{"data-target":"#pic-1","data-toggle":"tab"},n.a.createElement("img",{src:"https://bootstrapmade.com/demo/themes/EstateAgency/assets/img/slide-2.jpg"}))),n.a.createElement("li",null,n.a.createElement("a",{"data-target":"#pic-2","data-toggle":"tab"},n.a.createElement("img",{src:"https://bootstrapmade.com/demo/themes/EstateAgency/assets/img/slide-2.jpg"}))),n.a.createElement("li",null,n.a.createElement("a",{"data-target":"#pic-3","data-toggle":"tab"},n.a.createElement("img",{src:"https://bootstrapmade.com/demo/themes/EstateAgency/assets/img/slide-2.jpg"}))),n.a.createElement("li",null,n.a.createElement("a",{"data-target":"#pic-4","data-toggle":"tab"},n.a.createElement("img",{src:"https://bootstrapmade.com/demo/themes/EstateAgency/assets/img/slide-2.jpg"}))),n.a.createElement("li",null,n.a.createElement("a",{"data-target":"#pic-5","data-toggle":"tab"},n.a.createElement("img",{src:"https://bootstrapmade.com/demo/themes/EstateAgency/assets/img/slide-2.jpg"})))))}function i(e){return n.a.createElement("div",{className:"rating"},n.a.createElement("div",{className:"stars"},e.stars>=1?n.a.createElement("span",{className:"fa fa-star checked"}):n.a.createElement("span",{className:"fa fa-star"}),e.stars>=2?n.a.createElement("span",{className:"fa fa-star checked"}):n.a.createElement("span",{className:"fa fa-star"}),e.stars>=3?n.a.createElement("span",{className:"fa fa-star checked"}):n.a.createElement("span",{className:"fa fa-star"}),e.stars>=4?n.a.createElement("span",{className:"fa fa-star checked"}):n.a.createElement("span",{className:"fa fa-star"}),e.stars>=5?n.a.createElement("span",{className:"fa fa-star checked"}):n.a.createElement("span",{className:"fa fa-star"})),"undefined"!==typeof e.numReviews&&n.a.createElement("span",{className:"review-no"},e.numReviews," reviews"))}function d(e){return n.a.createElement("button",{className:"add-to-cart btn btn-default",type:"button"},"Delete")}function p(e){return n.a.createElement("button",{className:"add-to-cart btn btn-default",type:"button"},"Modify")}function E(e){return n.a.createElement("div",{className:"details col-md-6"},n.a.createElement(n.a.Fragment,null,n.a.createElement("h3",{className:"product-title"},"304 Blaster Up"),n.a.createElement("p",{className:"product-description"},"Chicago, IL 606543"),n.a.createElement(i,{stars:"2",numReviews:"99"}),n.a.createElement("p",{className:"product-description"},"A property description is the written portion of a real estate listing that describes the real estate for sale or lease. Nowadays, most buyers begin their property search online. Therefore, real estate descriptions are your best chance to sway buyers and sellers."),n.a.createElement("h4",{className:"price"},"current price: ",n.a.createElement("span",null,"$1800")),n.a.createElement("p",{className:"vote"},n.a.createElement("strong",null,"Property ID")," 15599"),n.a.createElement("p",{className:"vote"},n.a.createElement("strong",null,"Location")," US"),n.a.createElement("p",{className:"vote"},n.a.createElement("strong",null,"Property Type")," House"),n.a.createElement("p",{className:"vote"},n.a.createElement("strong",null,"Status")," Rent"),n.a.createElement("div",{className:"action"},n.a.createElement(n.a.Fragment,null,n.a.createElement(d,null),n.a.createElement(p,null)))))}function g(e){return n.a.createElement(n.a.Fragment,null,n.a.createElement("div",{className:"container"},n.a.createElement("div",{className:"card"},n.a.createElement("div",{className:"container-fliud"},n.a.createElement("div",{className:"wrapper row"},n.a.createElement(n.a.Fragment,null,n.a.createElement(o,null),n.a.createElement(E,null)))))))}var u=function(e){return n.a.createElement(g,null)};function h(){return n.a.createElement("button",{onClick:function(e){c.a.render(n.a.createElement(n.a.StrictMode,null,n.a.createElement(u,null)),document.getElementById("root"))}},"Detail")}var f=function(){return n.a.createElement("div",{className:"App"},n.a.createElement("header",{className:"App-header"},n.a.createElement("img",{src:m.a,className:"App-logo",alt:"logo"}),n.a.createElement("p",null,"Edit ",n.a.createElement("code",null,"src/App.js")," and save to reload."),n.a.createElement("p",null,"Flask Token = ",window.token),n.a.createElement("a",{className:"App-link",href:"https://reactjs.org",target:"_blank",rel:"noopener noreferrer"},"Learn React"),n.a.createElement(h,null)))};Boolean("localhost"===window.location.hostname||"[::1]"===window.location.hostname||window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/));c.a.render(n.a.createElement(n.a.StrictMode,null,n.a.createElement(f,null)),document.getElementById("root")),"serviceWorker"in navigator&&navigator.serviceWorker.ready.then((function(e){e.unregister()})).catch((function(e){console.error(e.message)}))}],[[4,1,2]]]);
//# sourceMappingURL=main.9e5ad9aa.chunk.js.map