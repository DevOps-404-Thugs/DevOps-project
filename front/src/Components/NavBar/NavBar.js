import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import Axios from 'axios'
import "./NavBar.css"


function NavBar() {
	const [loginState, setLoginState] = useState(0);//default as unlogin

	useEffect(() => {
		Axios.get(`http://127.0.0.1:8000/login`,{withCredentials: true})
		.then(response => {
			console.log("loginGET:"+response.status)
			console.log(response)
			if(response.status === 200){
				//console.log("loginState loginGET200:" + loginState);
				setLoginState(1)//unlogin
				//console.log("loginState after loginGET200:" + loginState);
			}else if(response.status === 205){
				setLoginState(0)//unlogin
				//console.log("loginState after loginGET205:" + loginState);
			}
		})
		// .catch(function(error){
		// 	console.log("loginGET:"+error.response.status)
		// 	console.log(error.response)
			
		// })

		console.log("loginState:" + loginState);
	},[]);
	

	const onLogout = (event) => {
		Axios.get(`http://127.0.0.1:8000/logout`, {withCredentials: true})
		.then(response => {
			console.log("logoutGET:"+response.status)
			console.log(response)
			if(response.status === 200){
				setLoginState(0)//unlogin
				//console.log("loginState after logout200:" + loginState);
			}
		})
		// .catch(function(error){
		// 	console.log("logoutGET:"+error.response.status)
		// 	console.log(error.response)
		// 	//console.log("loginState:" + loginState);
		// })
	}

  return (
    <nav class="menu">
			<div class="menu_container">
				<div class="menu_right">
					{loginState === 0?
						<Link class="link" to={'/login'}>Login</Link>
						: 
						<button class="btn btn-primary" onClick={onLogout}>log out</button> 
					}
						
				</div>
				
				<div class="menu_left">
					<Link class="link" to={'/'}>Home</Link>
				</div>
			</div>
			{/* <p>{loginState}</p> */}
    </nav>
  )
}

export default NavBar