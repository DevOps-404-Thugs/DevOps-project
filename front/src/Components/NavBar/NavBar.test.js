import React from 'react';
import renderer from 'react-test-renderer';
import {NavBar} from './NavBar.js';

test('Description generates correct page', () => {
    const component = renderer.create(
        <NavBar />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
