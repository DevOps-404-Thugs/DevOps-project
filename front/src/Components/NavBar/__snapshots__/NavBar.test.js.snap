// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description generates correct page 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <a
    className="navbar-brand"
    href="/ihomie"
  >
    iHomie
  </a>
  <button
    aria-controls="basic-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="basic-navbar-nav"
  >
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rb-event-key="/ihomie"
        href="/ihomie"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Home
      </a>
      <a
        className="nav-link"
        data-rb-event-key="/ihomie#/upload"
        href="/ihomie#/upload"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Upload
      </a>
      <a
        className="nav-link"
        data-rb-event-key="/ihomie#/register"
        href="/ihomie#/register"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Register
      </a>
    </div>
    <form
      className="form-inline"
    >
      <button
        className="btn btn-outline-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Login
      </button>
    </form>
  </div>
</nav>
`;
